compiler/rustc_mir_transform/src/check_alignment.rs

```diff
@@ -236,7 +237,11 @@ fn insert_alignment_check<'tcx>(
                 required: Operand::Copy(alignment),
                 found: Operand::Copy(addr),
             }),
-            unwind: UnwindAction::Terminate,
+            unwind: if tcx.sess.panic_strategy() == PanicStrategy::Unwind {
```

Comment:
This feels like something we're going to regress on with the next similar check in MIR -- can we introduce another run of AbortUnwindingCalls or otherwise make this more general?

(I'm not super familiar with mir opts so not sure if that pass is very expensive).

Reply:
Currently this pass is the only pass that can add a new call, and given that this pass is only enabled with debug assertions, re-running `AbortUnwindingCalls` feels a bit unnecessary?

Reply:
can we run `AbortUnwindingCalls` after it (or rather, run the alignment checks before it)

Reply:
alternatively, MIR validation could check that no normal unwind actions are present on panic=abort

Reply:
UnwindAction::Terminate may be present in `-Cpanic=abort` when `C-unwind` is used so we couldn't just do it in validation.

Lifting the `CheckAlignment` pass might make sense.

Reply:
terminate existing with panic=unwind could happen, but other unwind actions with panic=abort shouldn't, right? that could be validated

Reply:
That's entirely orthogonal to this PR though.

Reply:
If I understood @saethlin correctly, this could be `Unreachable` always?

Either way there should be a comment explaining why this is unreachable. "Because we use the non-unwinding panic machinery" seems like a good answer to me; it is explicitly intended to not unwind so I have no issue with this non-local dependency.